code/controllers/subsystem/changelog.dm

```diff
@@ -0,0 +1,220 @@
+			winset(C, "rpane.changelog", "background-color=#40628a;text-color=#FFFFFF")
+		else
+			winset(C, "rpane.changelog", "background-color=none;text-color=#000000")
+		return
+
+	// If SQL is enabled but we aint ready, queue them up, and use the default style
+	if(!ss_ready)
+		startup_clients_button += C
+		if(C.prefs.toggles & UI_DARKMODE)
+			winset(C, "rpane.changelog", "background-color=#40628a;text-color=#FFFFFF")
+		else
+			winset(C, "rpane.changelog", "background-color=none;text-color=#000000")
+		return
+
+	// If we are ready, process the button style
+	if(C.prefs.lastchangelog != current_cl_timestamp)
+		winset(C, "rpane.changelog", "background-color=#bb7700;text-color=#FFFFFF;font-style=bold")
+		to_chat(C, "<span class='info'>Changelog has changed since your last visit.</span>")
+	else
+		if(C.prefs.toggles & UI_DARKMODE)
+			winset(C, "rpane.changelog", "background-color=#40628a;text-color=#FFFFFF")
+		else
+			winset(C, "rpane.changelog", "background-color=none;text-color=#000000")
+
+
+/datum/controller/subsystem/changelog/proc/OpenChangelog(client/C)
+	// If SQL is enabled but we aint ready, queue them up
+	if(!ss_ready)
+		startup_clients_open += C
+		to_chat(C, "<span class='notice'>The changelog system is still initializing. The changelog will open for you once it has initialized.</span>")
+		return
+
+	UpdatePlayerChangelogDate(C)
+	UpdatePlayerChangelogButton(C)
+
+	var/datum/browser/cl_popup = new(C.mob, "changelog", "Changelog", 700, 800)
+	cl_popup.set_content(changelogHTML)
+	cl_popup.open()
+
+/client/verb/changes()
+	set name = "Changelog"
+	set desc = "View the changelog."
+	set category = "OOC"
+	// Just invoke the actual CL thing
+	SSchangelog.OpenChangelog(src)
+
+// Helper to turn CL types into a fontawesome icon instead of an image
+/datum/controller/subsystem/changelog/proc/Text2Icon(text)
+	switch(text)
+		if("FIX")
+			return "<i class='fas fa-tools'></i>"
+		if("WIP")
+			return "<i class='fas fa-hard-hat'></i>"
+		if("TWEAK")
+			return "<i class='fas fa-sliders-h'></i>"
+		if("SOUNDADD")
+			return "<i class='fas fa-volume-up'></i>"
+		if("SOUNDDEL")
+			return "<i class='fas fa-volume-mute'></i>"
+		if("CODEADD")
+			return "<i class='fas fa-plus'></i>"
+		if("CODEDEL")
+			return "<i class='fas fa-minus'></i>"
+		if("IMAGEADD")
+			return "<i class='fas fa-folder-plus'></i>"
+		if("IMAGEDEL")
+			return "<i class='fas fa-folder-minus'></i>"
+		if("SPELLCHECK")
+			return "<i class='fas fa-font'></i>"
+		if("EXPERIMENT")
+			return "<i class='fas fa-exclamation-triangle'></i>"
+		// No default here because its not possible to be anything else unless the DB schema changes
+
+// This proc is the star of the show
+/datum/controller/subsystem/changelog/proc/GenerateChangelogHTML()
+	// Modify the code below to modify the header of the changelog
+	var/changelog_header = {"
+		<link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" rel="stylesheet">
```

Comment:
This probably needs to be local CSS, but for that I need to figure out how the asset caching system works

Reply:
I have discovered why this is needed.

TODO: Add fontawesome webfont files locally